quinn-proto/src/range_set.rs

```diff
@@ -8,6 +8,292 @@ use std::{
     },
 };
 
+use tinyvec::TinyVec;
+
+/// A set of u64 values optimized for long runs and random insert/delete/contains
+///
+/// `ArrayRangeSet` uses an array representation, where each array entry represents
+/// a range.
```

Comment:
Would be nice to include some of the higher-level documentation from the commit message here, that is, why we need a second implementation of `RangeSet` (and maybe also explicitly call out ack ranges as the goal). Maybe we should even change the name to `AckRangeSet`?